src/lib/operators/operator_expression.cpp

```diff
@@ -0,0 +1,77 @@
+#include "operator_expression.hpp"
+
+#include "constant_mappings.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/lqp_expression.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+std::shared_ptr<OperatorExpression> OperatorExpression::create_column(const ColumnID column_id,
+                                                                      const std::optional<std::string>& alias) {
+  auto expression = std::make_shared<OperatorExpression>(ExpressionType::Column);
+  expression->_column_id = column_id;
+  expression->_alias = alias;
+
+  return expression;
+}
+
+OperatorExpression::OperatorExpression(const std::shared_ptr<LQPExpression>& lqp_expression,
+                                       const std::shared_ptr<AbstractLQPNode>& node)
+    : BaseExpression<OperatorExpression>(lqp_expression->_type) {
+  /**
+   * Ugly, but anyway: Copy all fields from LQPExpression and transform ColumnOrigins to ColumnIDs
```

Comment:
"Ugly, but anyway" doesn't help anyone. Either it is has to be done like this for a reason or not.

Reply:
added explanation. Left the "ugly" part in because I want my reader to know that I agree with the sentiment he's likely having when seeing this